libres/lib/enkf/enkf_fs.cpp

```diff
@@ -432,7 +434,11 @@ static void enkf_fs_fread_cases_config(enkf_fs_type *fs) {
 
 static void enkf_fs_fread_state_map(enkf_fs_type *fs) {
     char *filename = enkf_fs_alloc_case_filename(fs, STATE_MAP_FILE);
-    state_map_fread(fs->state_map, filename);
+    try {
+        fs->state_map->read(filename);
+    } catch (const ert::read_error &) {
+        /* Exception is ignored. StateMap is reset */
```

Comment:
Wouldnt it be more readable to but the resetting here instead of where the exception is thrown?